src/dtx/dtx_resync.c

```diff
@@ -86,10 +86,16 @@ dtx_resync_commit(uuid_t po_uuid, struct ds_cont_child *cont,
 		dte[i].dte_xid = dre->dre_xid;
 		dte[i].dte_oid = dre->dre_oid;
 
-		rc = vos_dtx_add_cos(cont->sc_hdl, &dre->dre_oid, &dre->dre_xid,
-				     dre->dre_hash, crt_hlc_get(),
-				     dre->dre_intent == DAOS_INTENT_PUNCH ?
-				     true : false);
+		rc = vos_dtx_lookup_cos(cont->sc_hdl, &dre->dre_oid,
+					&dre->dre_xid, dre->dre_hash,
+					dre->dre_intent == DAOS_INTENT_PUNCH ?
+					true : false);
+		if (rc == -DER_NONEXIST)
+			rc = vos_dtx_add_cos(cont->sc_hdl, &dre->dre_oid,
+				&dre->dre_xid, dre->dre_hash, crt_hlc_get(),
+				dre->dre_intent == DAOS_INTENT_PUNCH ?
+				true : false);
+
```

Comment:
Hmm, if this happens during the leader waiting for the reply from other replicas, then later the leader will try to add this again to cos? 

Probably we should do all of those dtx status change in a single ULT.

Reply:
Hmm, if this patch is intend to commit all of those committable dtx, then dtx_status_handle() looks a bit confusing to me. 
Since it seems for those dtx, which are already in the committable list does not need to 
do dtx_check and dtx_check_committable(), and even for check_leader() 

Could you please optimize the code a bit?

Reply:
Related logic is re-org and optimized in the new patch.